clients/go/coreutils/extract_literal_test.go

```diff
@@ -101,4 +104,32 @@ func TestFetchLiteral(t *testing.T) {
 		_, err = ExtractFromLiteral(p)
 		assert.NotNil(t, err)
 	})
+
+	t.Run("Generic", func(t *testing.T) {
+		literalVal := map[string]interface{}{
+			"x": 1,
+			"y": "ystringvalue",
+		}
+		var literalType = &core.LiteralType{Type: &core.LiteralType_Simple{Simple: core.SimpleType_STRUCT}}
+		lit, err := MakeLiteralForType(literalType, literalVal)
+		assert.NoError(t, err)
+		extractedLiteralVal, err := ExtractFromLiteral(lit)
```

Comment:
Is there a reason why this returns a structpb.Struct, instead of a map? It seems asymmetrical to provide a map in, and receive a structpb.Struct on the other side of the roundtrip.

Reply:
Given the end goal is to support a DataClass as an input, is there a stronger type that can be used instead of Generic, like some kind of mapping type?

(Sorry, not totally across the Flyte type system :) )

Reply:
Please let me if the response here helps https://github.com/flyteorg/flyteidl/pull/176#issuecomment-853766895

Reply:
I still think ExtractFromLiteral should return a native Go map instead of structpb.Struct because I feel most use cases would require native types. That said, there's precedence for returning pb types in this function anyway, e.g. Timestamp and Duration, so let's keep this behaviour.